### DISCUSSION TITLE
Caide tester

### DISCUSSION THREAD

**Discussion by Spidy104** [slycelote/caide#49] [Q&A] [answered]
The example of the caide cpp-builder.sh fails when I run caide test.

**Comment by slycelote** [comment]
Please share screenshots of the error, steps to reproduce and your entire caide directory as a zip file.

**Reply by Spidy104** [reply]
[CaideProject.zip](https://github.com/slycelote/caide/files/7367325/CaideProject.zip
![caide](https://user-images.githubusercontent.com/65475136/137777272-cb663d12-9034-4598-9637-4cd33b1ed8bd.png)
Here is all things, u asked for.

**Comment by slycelote** [comment, marked as answer]
> To configure a builder for a programming language, add a setting build_and_run_tests, in the section for that language (e.g., [cpp]).

Yours is in [cpp_builder].

**Reply by slycelote** [reply]
Also, I don't know why you use bash when you're on Windows.

**Reply by Spidy104** [reply]
Ok, I don't know batch, umm, ok, lemme see, the [cpp] thingy

**Reply by Spidy104** [reply]
Still, it fails badly, on shifting it to [cpp]

**Reply by slycelote** [reply]
Please share screenshots of the error, steps to reproduce and your entire caide directory as a zip file...

**Reply by Spidy104** [reply]
Got the error caide doesn't support cygwin.

**Reply by slycelote** [reply]
Please share screenshots of the error, steps to reproduce and your entire caide directory as a zip file.

**Comment by Spidy104** [comment]
ok, the caide directory is same, The, the error was something like this, and btw, I use cygdrive.
```bash
Executing custom test runner: cpp
bash /cygdrive/d/CaideProject/builder.sh

/cygdrive/d/CaideProject/builder.sh: line 2: $'\r': command not found
/cygdrive/d/CaideProject/builder.sh: line 7: $'\r': command not found
: invalid optioneProject/builder.sh: line 9: set: -
set: usage: set [-abefhkmnptuvxBCHP] [-o option-name] [--] [arg ...]
/cygdrive/d/CaideProject/builder.sh: line 10: $'\r': command not found
/cygdrive/d/CaideProject/builder.sh: line 13: $'\r': command not found
/cygdrive/d/CaideProject/builder.sh: line 14: $'caide\r': command not found
/cygdrive/d/CaideProject/builder.sh: line 15: $'caide\r': command not found
/cygdrive/d/CaideProject/builder.sh: line 16: $'\r': command not found
/cygdrive/d/CaideProject/builder.sh: line 17: cd: $'\r/\r\r': No such file or directory
.cpp: No such file or directory
compilation terminated.
_test.cpp: No such file or directory
compilation terminated.
: No such file or directoryilder.sh: line 19: ./
Builder exit code 127
Build failed
```

**Reply by slycelote** [reply]
Pretty sure that you changed caide.ini so caide directory is **not** the same.
Regarding the error, have you tried running `bash d:/CaideProject/builder.sh` by yourself? You'll get the same error. Bash requires Linux-style line separators (\n, not \r\n).

**Reply by slycelote** [reply]
Also, I don't know what you're trying to achieve with builders here. Default code template already verifies tests for you.

**Reply by Spidy104** [reply]
To check my answers, ofc.

**Reply by slycelote** [reply]
Default code template (BFailingGrade_test.cpp) checks your answers.

**Reply by Spidy104** [reply]
For me, it doesn't, dunno why

**Reply by Spidy104** [reply]
You can have a check, if u want, removing that build_and_tests section in the caide.ini file.

**Reply by slycelote** [reply]
Please compile and run BFailingGrade_test.cpp and provide the output.

**Reply by Spidy104** [reply]
Just the BFailingGrade_test.cpp or both BFailingGrade,cpp & BFailingGrade_test.cpp

**Reply by Spidy104** [reply]
```cmd
sh: C:\caide\caide.exe: command not found
caide update_tests returned non-zero error code 32512
Running test case1
2

Running test case2
3

Running test case3
1

sh: C:\caide\caide.exe: command not found
```

**Reply by slycelote** [reply]
both of them

**Reply by slycelote** [reply]
Looks like you moved your caide directory after creating (from c:\caide to somewhere else). Try moving back or creating from scratch.

**Reply by Spidy104** [reply]
ok

**Comment by Spidy104** [comment]
Ye, the error was the same, feels like I gotta build an executable, that runs the commands in the builder. sh
